cnf-certification-test/platform/bootparams/bootparams.go

```diff
@@ -0,0 +1,146 @@
+	}
+	for key, mcVal := range mcKernelArgumentsMap {
+		if currentVal, ok := currentKernelArgsMap[key]; ok {
+			if currentVal != mcVal {
+				return fmt.Errorf("currentVal!=mcVal %s!=%s", currentVal, mcVal)
+			}
+		}
+		if grubVal, ok := grubKernelConfigMap[key]; ok {
+			if grubVal != mcVal {
+				return fmt.Errorf("currentVal!=mcVal %s!=%s", grubVal, mcVal)
+			}
+		}
+	}
+	return nil
+}
+
+func getMcName(env *provider.TestEnvironment, nodeName string) (mcName string, err error) {
+	if mcName, ok := env.Nodes[nodeName].Annotations["machineconfiguration.openshift.io/currentConfig"]; ok {
+		return mcName, nil
+	}
+	return "", fmt.Errorf("no machineconfiguration.openshift.io/currentConfig present in node=%s", nodeName)
+}
+
+func getMcKernelArguments(env *provider.TestEnvironment, mcName string) (aMap map[string]string, err error) {
+	o := clientsholder.GetClientsHolder()
+	mc, err := o.McClient.MachineconfigurationV1().MachineConfigs().Get(context.TODO(), mcName, metav1.GetOptions{})
+	if err != nil {
+		return aMap, fmt.Errorf("failed to get machine config object %s", mcName)
+	}
+	mcKernelArgumentsMap := ArgListToMap(mc.Spec.KernelArguments)
+	return mcKernelArgumentsMap, nil
+}
+
+func getGrubKernelArgs(env *provider.TestEnvironment, nodeName string) (aMap map[string]string, er error) {
+	o := clientsholder.GetClientsHolder()
+	ctx := clientsholder.Context{Namespace: env.DebugPods[nodeName].Namespace,
+		Podname:       env.DebugPods[nodeName].Name,
+		Containername: env.DebugPods[nodeName].Spec.Containers[0].Name}
+	bootConfig, errStr, err := o.ExecCommandContainer(ctx, grubKernelArgsCommand)
+	if err != nil || errStr != "" {
+		return aMap, fmt.Errorf("cannot exucute %s on debug pod %s, err=%s, stderr=%s", grubKernelArgsCommand, env.DebugPods[nodeName], err, errStr)
+	}
+
+	splitBootConfig := strings.Split(bootConfig, "\n")
+	filteredBootConfig := FilterArray(splitBootConfig, func(line string) bool {
+		return strings.HasPrefix(line, "options")
+	})
+	if len(filteredBootConfig) != 1 {
+		return aMap, fmt.Errorf("filteredBootConfig!=1")
+	}
+	grubKernelConfig := filteredBootConfig[0]
+	grubSplitKernelConfig := strings.Split(grubKernelConfig, " ")
+	grubSplitKernelConfig = grubSplitKernelConfig[1:]
+	return ArgListToMap(grubSplitKernelConfig), nil
+}
+
+func getCurrentKernelCmdlineArgs(cut *provider.Container) (aMap map[string]string, err error) {
+	currnetKernelCmdlineArgs, stderr, err := crclient.ExecCommandContainerNSEnter(kernelArgscommand, cut)
+	if err != nil || stderr != "" {
+		return aMap, fmt.Errorf("cannot exucute getGrubKernelArgs on container %s, err=%s, stderr=%s", cut, err, stderr)
+	}
+	currentSplitKernelCmdlineArgs := strings.Split(currnetKernelCmdlineArgs, " ")
+	return ArgListToMap(currentSplitKernelCmdlineArgs), nil
+}
+
+// ArgListToMap takes a list of strings of the form "key=value" and translate it into a map
+// of the form {key: value}
+func ArgListToMap(lst []string) map[string]string {
```

Comment:
This function is generic enough to be defined in some other common package, so that it can be shared by more test suites. Do we have such a place?

Reply:
fixed, added a folder /pkg/arrayhelper